modules/registry-refactoring/src/main/java/org/apache/airavata/registry/core/repositories/expcatalog/JobRepository.java

```diff
@@ -0,0 +1,181 @@
+package org.apache.airavata.registry.core.repositories.expcatalog;
+
+import org.apache.airavata.model.job.JobModel;
+import org.apache.airavata.model.status.JobStatus;
+import org.apache.airavata.model.task.TaskModel;
+import org.apache.airavata.registry.core.entities.expcatalog.JobEntity;
+import org.apache.airavata.registry.core.entities.expcatalog.JobStatusEntity;
+import org.apache.airavata.registry.core.utils.DBConstants;
+import org.apache.airavata.registry.core.utils.ExpCatalogUtils;
+import org.apache.airavata.registry.core.utils.ObjectMapperSingleton;
+import org.apache.airavata.registry.core.utils.QueryConstants;
+import org.apache.airavata.registry.cpi.CompositeIdentifier;
+import org.apache.airavata.registry.cpi.RegistryException;
+import org.dozer.Mapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class JobRepository extends ExpCatAbstractRepository<JobModel, JobEntity, String> {
+    private final static Logger logger = LoggerFactory.getLogger(JobRepository.class);
+
+    public JobRepository() { super(JobModel.class, JobEntity.class); }
+
+    protected String saveJobModelData(JobModel jobModel, CompositeIdentifier cis) throws RegistryException {
+        JobEntity jobEntity = saveJob(jobModel, cis);
+        return jobEntity.getJobId();
+    }
+
+    protected JobEntity saveJob(JobModel jobModel, CompositeIdentifier cis) throws RegistryException {
```

Comment:
Can we get rid of this CompositeIdentifier? I think since TASK_ID should be part of the key we can use JobEntityPK instead of CompositeIdentifier.